### PR TITLE
Cityview fixes

### PR DESCRIPTION
**Pull request by LauchInterceptor**
- fixed a problem where the element selection was offset
- improved performance of the autosuggest panels where you choose commits
- improved commit autosuggestion
- typeified a lot of ambiguous var's for better understandability

### CODE REVIEW

**Comment by maximAtanasov** [comment]
@LauchInterceptor I've fixed a small issue in the ColorHelper and applied the settings from TSLint to the code.  However there are a few issues for me. I cannot search for files anymore (Search box doesn't show or do anything) and the dropdown with all the commits only shows up if I delete the text of the commit that is already there. 

Also, would it be possible to add a tooltip to the opposite commit when hovering over a file in the split view? 

Another issue is that there are sometimes rendering artifacts when colors or geometry overlap eachother. Is it possible to fix these somehow?

**Comment by maximAtanasov** [comment]
@LauchInterceptor  is this PR ready for review?

**Comment by LauchInterceptor** [comment]
@LauchInterceptor There are still small bugs and improvements which aren't function critical though. The ones im aware of will be done by tomorrow evening.

**Comment by LauchInterceptor** [comment]
PR is now ready for review!

File: coderadar-ui/package.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
   "version": "0.2.3",
   "scripts": {
     "ng": "node --max_old_space_size=5048 ./node_modules/@angular/cli/bin/ng",
-    "start": "node --max_old_space_size=5048 ./node_modules/@angular/cli/bin/ng serve --prod",
+    "start": "node --max_old_space_size=5048 ./node_modules/@angular/cli/bin/ng serve",
+    "prodserve": "node --max_old_space_size=5048 ./node_modules/@angular/cli/bin/ng serve --prod --port 4201",
```
Comment (maximAtanasov) [inline coderadar-ui/package.json]: A separate "prod" command isn't a bad idea. However, running `npm prodserve` doesn't do anything for me (it just tells me the command doesn't exist). Also, why does it start on port 4201?

File: coderadar-ui/src/app/city-map/visualization/screen/screen.component.ts [hunk 2]
```diff
@@ -342,4 +351,45 @@ export class ScreenComponent implements OnInit, OnChanges, OnDestroy {
...
+      color:0xff0000
+    });
+    var geometry = new THREE.Geometry();
+    geometry.vertices.push(
+      new Vector3(0,0,0),
+      new Vector3(0,-1,0)
+    );
+    var tipSize = 0.1;
+    var tipGeometry = new THREE.SphereGeometry(tipSize,16,16);
+    /*tipGeometry.vertices.push(
```
Comment (maximAtanasov) [inline coderadar-ui/src/app/city-map/visualization/screen/screen.component.ts]: Commented-out code needed?

File: coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.scss [hunk 3]
```diff
@@ -10,20 +10,20 @@
...
   &.visible {
-    opacity: 1;
+    opacity: 0.8;
   }
 
   & > .element-name {
     text-overflow: ellipsis;
     text-align: left;
-    white-space: nowrap;
+    //white-space: nowrap;
```
Comment (maximAtanasov) [inline coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.scss]: still needed?

File: coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.ts [hunk 4]
```diff
@@ -54,8 +56,10 @@ export class TooltipComponent implements OnInit {
   }
 
   followPosition(position: { x: number, y: number }) {
-    this.tooltipElement.style.left = position.x + 15 + 'px';
-    this.tooltipElement.style.top = position.y + 15 + 'px';
+    // this.tooltipElement.style.left = position.x + 15 + 'px';
```
Comment (maximAtanasov) [inline coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.ts]: still needed?

**Comment by maximAtanasov** [comment]
Everything works as it should. I'll create issues for the other needed improvements after we merge this.

File: coderadar-ui/package.json (hunk 1 above)
Comment (LauchInterceptor) [reply coderadar-ui/package.json]: I added the script so I could have both a production and a debug server running.
We can remove this for now and discuss a solution if necessary.

File: coderadar-ui/src/app/city-map/visualization/screen/screen.component.ts (hunk 2 above)
Comment (LauchInterceptor) [reply coderadar-ui/src/app/city-map/visualization/screen/screen.component.ts]: Not needed

File: coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.ts (hunk 4 above)
Comment (LauchInterceptor) [reply coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.ts]: Not needed

File: coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.scss (hunk 3 above)
Comment (LauchInterceptor) [reply coderadar-ui/src/app/city-map/visualization/tooltip/tooltip.component.scss]: Not needed

**Pull request merged**